### PR TITLE
Tutorial v0.31 update

### PR DESCRIPTION
**Pull request by keitht547**
tutorial update to include gapfill and fastgapfill. Also updates section related to new model import
with psamm-import.

### CODE REVIEW

File: docs/tutorial/curation.rst [hunk 1]
```diff
@@ -473,11 +473,130 @@ a network based optimization to identify metabolites with no production pathways
...
+
+FastGapFill
+~~~~~~~~~~~
+
+The ``fastgapfill`` function in `PSAMM` is different gap-filling method that uses the FastGapFill algorithm
+to attempt to generate a gap-filled model that is entirely flux consistent [Thiele14]_. The implementation
+of this algorithm in `PSAMM` can be utilized for unblocking an entire metabolic model or for unblocking
+specific reactions in a network. Often times unblocking all of the reactions in a model at the same time
+will not produce the most meaningful and easy to understand results so only performing this function on a
+subset of reactions is preferable. To do this the ``--subset`` option can be used to provide a file that
```
Comment (jonls) [inline docs/tutorial/curation.rst]: Just a thought for a future improvement: Maybe this should work like `--compound` in `gapfill` where the compounds can be specified directly, or in a file with the `@`-prefix?

File: docs/tutorial/curation.rst [hunk 2]
```diff
@@ -473,11 +473,130 @@ a network based optimization to identify metabolites with no production pathways
...
+contains a list of reactions to unblock. In this example that list would look like this:
+
+.. code-block:: shell
```
Comment (jonls) [inline docs/tutorial/curation.rst]: This is not a command so should not be a `shell` code block. (just change "...would look like this:" to double colon "...would look like this::" and remove the ".. code-block:: shell" line.)

File: docs/tutorial/curation.rst [hunk 3]
```diff
@@ -473,11 +473,130 @@ a network based optimization to identify metabolites with no production pathways
...
+model:
+
+
+.. code-block:: shell
+
+    (psamm-env) $ psamm-model fastgapfill --subset subset.tsv
+
+In this case the output from this command will show the following:
+
+.. code-block:: shell
```
Comment (jonls) [inline docs/tutorial/curation.rst]: Not `shell`.

File: docs/tutorial/curation.rst [hunk 4]
```diff
@@ -473,11 +473,130 @@ a network based optimization to identify metabolites with no production pathways
...
+    EX_c[e]	Add	1	c[e] <=>
+    EX_d[e]	Add	1	d[e] <=>
+    EX_succ_c[e]	Add	1	Succinate[e] <=>
+    TP_c[c]_c[e]	Add	1	c[c] <=> c[e]
+    TP_d[c]_d[e]	Add	1	d[c] <=> d[e]
+    TP_succ_c[c]_succ_c[e]	Add	1	Succinate[c] <=> Succinate[e]
+    rxn2	Add	1	a[c] => b[c]
+
+The output will first list the model reactions which are labeled with the 'Model' tag in the second column
+of the output. The `PSAMM` will list out any artificial exchange and transporters, as well as any gap reactions
```
Comment (jonls) [inline docs/tutorial/curation.rst]: Just "`PSAMM` will list..."?

File: docs/tutorial/curation.rst [hunk 5]
```diff
@@ -473,11 +473,130 @@ a network based optimization to identify metabolites with no production pathways
...
+included from the larger database. These will be labeled with the `Add` tag in the second column. When compared
+to the ``gapfill`` results from the previous section it can be seen that the ``fastgapfill`` result suggests
+some artificial transporters and exchange reactions for certain compounds. this is due to this method
```
Comment (jonls) [inline docs/tutorial/curation.rst]: this -> This

File: docs/tutorial/import_export.rst [hunk 6]
```diff
@@ -36,11 +36,24 @@ The ``psamm-import`` program supports the import of models in various formats.
...
-the support for importing from Excel file is model specific and are available
-for 17 published models. There is also a generic Excel import for models
-produced by the ModelSEED pipeline. To see a list of these models or model
-formats that are supported, use the command:
+`BiGG`_ database or from locally downloaded versions.
+
+The support for importing from Excel file is model specific and are available
+for 17 published models. This import requires the installation of the separate
+psamm-import repository. There is also a generic Excel import for models
+produced by the ModelSEED pipeline.
```
Comment (jonls) [inline docs/tutorial/import_export.rst]: Maybe mention that this is for the old ModelSEED? It doesn't work for the new ModelSEED as far as I remember.

File: docs/tutorial/import_export.rst [hunk 7]
```diff
@@ -36,11 +36,24 @@ The ``psamm-import`` program supports the import of models in various formats.
...
+
+To install the ``psamm-import`` package for excel format models use the following
```
Comment (jonls) [inline docs/tutorial/import_export.rst]: excel -> Excel

File: docs/tutorial/import_export.rst [hunk 8]
```diff
@@ -36,11 +36,24 @@ The ``psamm-import`` program supports the import of models in various formats.
...
+command:
+
+.. code-block:: shell
+
+    (psamm-env) $ pip install git+https://github.com/zhanglab/psamm-import.git
+
+This install will make the excel importers available from the command line when the
```
Comment (jonls) [inline docs/tutorial/import_export.rst]: excel -> Excel

File: docs/tutorial/curation.rst (hunk 1 above)
Comment (keitht547) [inline docs/tutorial/curation.rst]: i agree. the idea behind the subset of reactions to make consistent could be confusing when compared to how gapfill works.

File: docs/tutorial/curation.rst [hunk 9]
```diff
@@ -473,11 +473,126 @@ a network based optimization to identify metabolites with no production pathways
...
+functions ``gapfill``, ``fastgapfill``, and ``completepath`` can be used to perform these gapfilling procedures
 during the process of generating and curating a model.
 
+GapFill
+~~~~~~~
+The ``gapfill`` function in PSAMM can be used to apply a GapFill algorithm based on [Kumar07]_ to a metabolic model
+to search for and identify reactions that can be added into a model to unblock the production of a specific
+compound or set of compounds. To provide an example of how to utilize this ``gapfill`` function a version of
+the E. coli core model has been provided in the `tutorial-part-2/Gapfilling_Model/` directory. In this directory
+is the E. coli core model with a small additional, incomplete pathway, added that contains the following reactions:
```
Comment (jonls) [inline docs/tutorial/curation.rst]: @keitht547 Was this added to `psamm-tutorial` already?

File: docs/tutorial/import_export.rst [hunk 10]
```diff
@@ -36,11 +36,25 @@ The ``psamm-import`` program supports the import of models in various formats.
...
+current ModelSEED can be imported in the SBML format.
+
+To install the ``psamm-import`` package for Excel format models use the following
+command:
+
+.. code-block:: shell
+
+    (psamm-env) $ pip install git+https://github.com/zhanglab/psamm-import.git
+
+This install will make the Ezxcel importers available from the command line when the
```
Comment (jonls) [inline docs/tutorial/import_export.rst]: "Excel" 😛

**Pull request merged**